Source/OxyPlot/Foundation/DataVector.cs

```diff
@@ -0,0 +1,199 @@
+        }
+
+        /// <summary>
+        /// Subtracts one specified vector from another.
+        /// </summary>
+        /// <param name="v">The vector to subtract from.</param>
+        /// <param name="d">The vector to be subtracted.</param>
+        /// <returns>The result of operation.</returns>
+        public static DataVector operator -(DataVector v, DataVector d)
+        {
+            return new DataVector(v.x - d.x, v.y - d.y);
+        }
+
+        /// <summary>
+        /// Negates the specified vector.
+        /// </summary>
+        /// <param name="v">The vector to negate.</param>
+        /// <returns>The result of operation.</returns>
+        public static DataVector operator -(DataVector v)
+        {
+            return new DataVector(-v.x, -v.y);
+        }
+
+        /// <summary>
+        /// Normalizes this vector inplace.
+        /// </summary>
+        /// <remarks>
+        /// If the vector has zero length, then it is left unchanged.
+        /// </remarks>
+        public void Normalize()
+        {
+            double l = Math.Sqrt((this.x * this.x) + (this.y * this.y));
+            if (l > 0)
+            {
+                this.x /= l;
+                this.y /= l;
+            }
+        }
+
+        /// <summary>
+        /// Returns C# code that generates this instance.
+        /// </summary>
+        /// <returns>The to code.</returns>
+        public string ToCode()
+        {
+            return CodeGenerator.FormatConstructor(this.GetType(), "{0},{1}", this.x, this.y);
+        }
+
+        /// <summary>
+        /// Determines whether this and another specified <see cref="T:DataVector" /> have the same value.
+        /// </summary>
+        /// <param name="other">The vector to compare to this instance.</param>
+        /// <returns><c>true</c> if the value of the <paramref name="other" /> parameter is the same as the value of this instance; otherwise, <c>false</c>.</returns>
+        public bool Equals(DataVector other)
+        {
+            return this.x.Equals(other.x) && this.y.Equals(other.y);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="System.String" /> that represents this instance.
+        /// </summary>
+        /// <returns>A <see cref="System.String" /> that represents this instance.</returns>
+        public override string ToString()
+        {
+            return this.x + " " + this.y;
+        }
+
+        /// <summary>
+        /// Determines whether this point is defined.
+        /// </summary>
+        /// <returns><c>true</c> if this point is defined; otherwise, <c>false</c>.</returns>
+        public bool IsDefined()
+        {
+            // check that x and y is not NaN (the code below is faster than double.IsNaN)
+#pragma warning disable 1718
+            // ReSharper disable EqualExpressionComparison
+            // ReSharper disable CompareOfFloatsByEqualityOperator
+            return this.x == this.x && this.y == this.y;
```

Comment:
I am not quite sure if that comparison is actually faster than `double.NaN` (anymore?), however I am fairly sure that it wouldn't really matter in terms of magnitude even if there was a difference... so I think I would prefer `double.NaN` here to avoid all the surrounding clutter.